azkaban-common/src/main/java/azkaban/executor/ExecutionReference.java

```diff
@@ -18,16 +18,19 @@
 
 public class ExecutionReference {
   private final int execId;
-  private final String host;
-  private final int port;
+  private Executor executor;
   private long updateTime;
   private long nextCheckTime = -1;
   private int numErrors = 0;
 
-  public ExecutionReference(int execId, String host, int port) {
+
+  public ExecutionReference(int execId) {
+    this.execId = execId;
+  }
+
+  public ExecutionReference(int execId, Executor executor) {
     this.execId = execId;
-    this.host = host;
-    this.port = port;
+    this.executor = executor;
```

Comment:
Should we check for null because we will reference it later - executor.getHost(), executor.getPort()?